cli/command/stack/deploy_composefile.go

```diff
@@ -321,7 +321,16 @@ func deployServices(
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
 			}
-
+			if !updateOpts.QueryRegistry {
+				serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
```

Comment:
I think the intent here would be more clear as an `else` instead of checking the flag set in the previous branch.

An inline comment about why this is necessary would also be great. I'm sure I'm going to forget. Something like this:

> If the image hasn't changed the serviceSpec needs to be updated to use the existing image that was set by `QueryRegistry` on the previous deploy. Otherwise this will trigger an incorrect service update.